Sample for chrome.management #1102 (#1105)
GoogleChrome/chrome-extensions-samples: api-samples/management/README.md

# chrome.management Extension

The [`chrome.management`](https://developer.chrome.com/docs/extensions/reference/api/management) API provides ways to manage the list of extensions/apps that are installed and running. It is particularly useful for extensions that [override](https://developer.chrome.com/extensions/develop/ui/override-chrome-pages) the built-in New Tab page.

## Features

![Screenshot](images/screenshot.png)

- List all installed extensions.
- Display extension icons, names, and versions.
- Provide an "Uninstall" button for each extension.

## Installation

1. Clone or download this repository.
2. Load the extension in Chrome as an [unpacked extension](https://developer.chrome.com/docs/extensions/mv3/getstarted/development-basics/#load-unpacked).

## Usage

1. Click on the extension icon to open the extension manager.
2. The manager will list all installed extensions.
3. Each extension entry includes its icon, name, and version.
4. Click the "Uninstall" button next to an extension to remove it.

## Code Overview

### `popup.js`

- Uses `chrome.management.getAll` to get a list of all installed extensions.

> - Creates a list item for each extension, including its icon, name, version, and an "Uninstall" button.

Reviewer: ```suggestion
- Creates a list item for each extension, including its icon, name, version, and an **Uninstall** button.
```

> - Sends a message to background script (`background.js`) to uninstall an extension when the "Uninstall" button is clicked.

Reviewer: ```suggestion
- Sends a message to the extension service worker (`background.js`) to uninstall an extension when the **Uninstall** button is clicked.
```

### `background.js`

- Listens for messages from the popup.

> - Handles uninstallation requests by calling `chrome.management.uninstall` with the extension id.

Reviewer: ```suggestion
- Handles uninstallation requests by calling `chrome.management.uninstall()` with the extension ID.
```
